src/microbe_stage/editor/MicrobeEditorGUI.cs

```diff
@@ -557,33 +666,106 @@ internal void OnMouseEnter()
     }
 
     /// <summary>
-    ///   Called when the mouse is no longer hovering
-    ///   the editor GUI.
+    ///   Called when the mouse is no longer hovering the editor GUI.
     /// </summary>
     internal void OnMouseExit()
     {
-        editor.ShowHover = inEditorTab;
+        editor.ShowHover = selectedEditorTab == EditorTab.CellEditor;
     }
 
+    /// <summary>
+    ///   Used by the selection options on the selection menu to show tooltips
+    /// </summary>
     internal void OnItemMouseHover(string itemName)
     {
-        foreach (PanelContainer tooltip in itemTooltipElements)
+        tooltipHandler.MainTooltip = tooltipHandler.GetTooltip(itemName);
+        tooltipHandler.Display = true;
+    }
+
+    /// <summary>
+    ///   Shows tooltips for the production balance bar
+    /// </summary>
+    internal void OnAtpProductionBarMouseEnter(string name)
+    {
+        if (energyBalanceInfo == null)
+            return;
+
+        var displayName = SimulationParameters.Instance.GetOrganelleType(name).Name;
+        var value = energyBalanceInfo.Production[name];
+
+        tooltipHandler.ShowDefaultTooltip($"{displayName}: +{value} ATP", 0.1f);
+    }
+
+    /// <summary>
+    ///   Shows tooltips for the consumption balance bar
+    /// </summary>
+    internal void OnAtpConsumptionBarMouseEnter(string name)
+    {
+        if (energyBalanceInfo == null)
+            return;
+
+        string displayName;
+        var value = energyBalanceInfo.Consumption[name];
+
+        switch (name)
         {
-            tooltip.Hide();
+            case "osmoregulation":
+            {
+                displayName = "Osmoregulation";
+                break;
+            }
+
+            case "baseMovement":
+            {
+                displayName = "Base Movement";
+                break;
+            }
 
-            if (tooltip.Name == itemName)
+            default:
             {
-                tooltip.Show();
+                displayName = SimulationParameters.Instance.GetOrganelleType(name).Name;
```

Comment:
Same thing here with the type.

Reply:
That's just for getting the display name to show on the atp balance tooltip, `name` string is an InternalName

Reply:
Okay. On second reading this code looks fine.